swift/ql/lib/codeql/swift/elements/expr/ApplyExpr.qll

```diff
@@ -9,6 +9,10 @@ class ApplyExpr extends ApplyExprBase {
     result = "call to " + this.getStaticTarget().toString()
     or
     not exists(this.getStaticTarget()) and
+    result = "call to " + this.getFunction().toString()
+    or
+    not exists(this.getStaticTarget()) and
+    not exists(this.getFunction()) and
```

Comment:
If there is no function then there's definitely no static call target. So the `not exists(this.getStaticTarget())` conjunction seems unnecessary.

Really, there shouldn't be any case where where `getFunction` doesn't have a result.

Also, I'm not really a big fan of the strings produced in the example below. The string "call to call to -(_:)" doesn't look like something I'd like to see in an alert message (which is the main point of `toString` in my opinion)

Reply:
Hmm, I hadn't noticed the repetition of `call to` actually.  I'll see if I can do better.

I agree about `getStaticTarget`.

Reply:
Done the latter, still looking at the former.